hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/map/MapKeySetWithPagingPredicateMessageTask.java

```diff
@@ -42,12 +42,12 @@ public MapKeySetWithPagingPredicateMessageTask(ClientMessage clientMessage, Node
 
     @Override
     protected Object reduce(Collection<QueryResultRow> result) {
-        PagingPredicateImpl pagingPredicate = (PagingPredicateImpl) getPredicate();
-        List<Map.Entry<Data, Data>> entriesData = getSortedPageEntries(result, pagingPredicate);
+        List<Map.Entry<Data, Data>> entriesData = getSortedPageEntries(result);
```

Comment:
We are still creating two lists here (`entriesData` and `keyList`), which seems unnecessary. Can it be avoided?

Reply:
Done.

Reply:
I see we still create a separate `keyList` list from the result of `getSortedPageEntries()` but let's leave it as it is. This can be optimized later if it proves to be a bottleneck.